src/Equinox.EventStoreDb/EventStoreDb.fs

```diff
@@ -181,36 +188,43 @@ module private Read =
             "Esdb{action:l} stream={stream} count={count}/{batches} version={version}",
             action, streamName, count, batches, version)
     let private loadBackwardsUntilOrigin (log : ILogger) (conn : EventStoreClient) batchSize streamName (tryDecode, isOrigin) ct
-        : Task<int64 * struct (ResolvedEvent * 'event voption)[]> = task {
+        : Task<int64 * struct (ResolvedEvent * 'event voption)[]> =
         let res = conn.ReadStreamAsync(Direction.Backwards, streamName, StreamPosition.End, int64 batchSize, resolveLinkTos = false, cancellationToken = ct)
-        try let! events =
-                res
-                |> TaskSeq.map (fun x -> struct (x, tryDecode x))
-                |> TaskSeq.takeWhileInclusive (function
-                    | x, ValueSome e when isOrigin e ->
-                        log.Information("EsdbStop stream={stream} at={eventNumber}", streamName, let en = x.Event.EventNumber in en.ToInt64())
-                        false
-                    | _ -> true)
-                |> TaskSeq.toArrayAsync
-            let v = match Seq.tryHead events with Some (r, _) -> let en = r.Event.EventNumber in en.ToInt64() | None -> -1
-            Array.Reverse events
-            return v, events
-        with :? AggregateException as e when (e.InnerExceptions.Count = 1 && e.InnerExceptions[0] :? StreamNotFoundException) ->
-            return -1L, [||] }
+        let processResult (result: EventStoreClient.ReadStreamResult) =
+            task {
```

Comment:
for better or for worse, `task {` is always at the end of the preceding line in this repo